libs/gltfio/src/ResourceLoader.cpp

```diff
@@ -1005,27 +1016,99 @@ void ResourceLoader::updateBoundingBoxes(FFilamentAsset* asset) const {
+    const size_t skinningAttrSize = cgltf_num_components(cgltf_type_vec4);
+    auto computeBoundingBoxSkinned = [&](const cgltf_primitive* prim, const Skin* skin, Aabb* result) {
+        Aabb aabb;
+        std::vector<mat4f> inverseGlobalTransforms(skin->targets.size());
+        for (size_t i = 0; i < skin->targets.size(); i++) {
+            auto xformable = tm.getInstance(skin->targets[i]);
+            if (xformable) {
+                inverseGlobalTransforms[i] = inverse(tm.getWorldTransform(xformable));
+            }
+        }
+        std::vector<float> verts;
+        std::vector<float> rawJoints;
+        std::vector<float> weights;
+        for (cgltf_size slot = 0; slot < prim->attributes_count; slot++) {
+            const cgltf_attribute& attr = prim->attributes[slot];
+            const cgltf_accessor* accessor = attr.data;
+            if (attr.type == cgltf_attribute_type_position && cgltf_num_components(accessor->type) >= posAttrSize) {
+                verts.resize(accessor->count * posAttrSize);
+                cgltf_accessor_unpack_floats(accessor, &verts[0], accessor->count * posAttrSize);
+            }
+            if (attr.type == cgltf_attribute_type_joints && cgltf_num_components(accessor->type) >= skinningAttrSize) {
+                rawJoints.resize(accessor->count * skinningAttrSize);
+                cgltf_accessor_unpack_floats(accessor, &rawJoints[0], accessor->count * skinningAttrSize);
+            }
+            if (attr.type == cgltf_attribute_type_weights && cgltf_num_components(accessor->type) >= skinningAttrSize) {
+                weights.resize(accessor->count * skinningAttrSize);
+                cgltf_accessor_unpack_floats(accessor, &weights[0], accessor->count * skinningAttrSize);
+            }
+        }
+        std::vector<size_t> jointIndices(rawJoints.begin(), rawJoints.end());
+        auto primitiveCount = static_cast<size_t>(verts.size() / posAttrSize);
+        for (size_t i = 0; i < primitiveCount; i++) {
+            float3 point(verts[posAttrSize * i], verts[posAttrSize * i + 1], verts[posAttrSize * i + 2]);
+            mat4f tmp = mat4f(0.0f);
+            for (size_t j = 0; j < skinningAttrSize; j++) {
+                size_t jointIndex = jointIndices[skinningAttrSize * i + j];
+                float jointWeight = weights[skinningAttrSize * i + j];
+                Entity jointEntity = skin->joints[jointIndex];
+                mat4f globalJointTransform = tm.getWorldTransform(tm.getInstance(jointEntity));
+                mat4f inverseBindMatrix = skin->inverseBindMatrices[jointIndex];
+                tmp += jointWeight * globalJointTransform * inverseBindMatrix;
+            }
+            for (const auto& inverseGlobalTransform: inverseGlobalTransforms) {
+                mat4f skinMatrix = inverseGlobalTransform * tmp;
+                if (!pImpl->mNormalizeSkinningWeights) {
+                    skinMatrix /= skinMatrix[3].w;
+                }
```

Comment:
@Ankbzpx why do we care about weight normalization here?  Weight normalization is a separate feature of the loader and it occurs earlier in the pipeline.

Reply:
Here was my take:

The `skinMatrix` is the weighted sum of each `jointMatrice`. If the weights were not normalized, the weighted sum of the `m[3].w` (the last element) of `jointMatrics` is not guaranteed to be 1, so the resulting transformation matrix `skinMatrix` needs to be normalized.

Reply:
Thanks, I understand now, but the glTF spec says that they should add as close to 1.0 as possible (section 3.7.3.3).  We added the `mNormalizeSkinningWeights` feature only for non-compliant assets.  If include this `/= w` thing, then we're adding yet another accommodation for non-complaint assets, so I don't think it's necessary.  Moreover it does not match what Filament's vertex shader does.

Reply:
Ah, I see. The logic does sound. Thanks for clarifying and please help remove it then

Reply:
By the way @Ankbzpx , is the AABB computation feature something that you're actively using?  It defaults to off and I'd be surprised if many folks are using it.

I'm asking because it is posing some challenges with some upcoming features, and we are considering removing this feature.

Reply:
I no longer work there so I am not sure anymore. Here was our use case back then:

We occasionally received animated `FBX` models authored on client side involving many modeling tricks, that when converted to `glb`, had non identity bind transforms (i.e. in Blender, model shown in the `Object` mode and `Edit` mode varies substantially). Due to nested hierarchy and animation, they were non-trivial to fix in modeling software (it's possible to bake inverse bind transform directly onto geometry, but at risk of breaking it). Thus, this PR aimed to relax those situations.

I believe it falls into category of accommodating non-complaint assets, and since it also hinders your development, it would be logical to remove it for now.